### PR TITLE
Add docs for integrating Blobstream

### PR DESCRIPTION
**Pull request by adlerjohn**
## Overview

Adds documentation for integrating with Blobstream contracts and off-chain components. Ref: https://github.com/celestiaorg/blobstream-contracts

Page previews in this PR:
https://docs.celestia.org/pr-preview/pr-1152/developers/blobstream/
https://docs.celestia.org/pr-preview/pr-1152/developers/blobstream-contracts/
https://docs.celestia.org/pr-preview/pr-1152/developers/blobstream-offchain/

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: docs/developers/blobstream-contracts.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        // Everything checks out, append rollup block hash to list.
+        rollup_block_hashes.push(_rollup_block_hash);
+    }
+
+    function verifyZKP(
+        bytes calldata _zk_proof,
+        bytes32 _data_root
+    ) private pure returns (bool) {
+        return true;
+    }
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:64]: Is  this supposed to make a call to Blobstream X?

File: docs/developers/blobstream-contracts.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+}
+```
+
+## Data Structures
+
+Each `DataRootTuple` is a tuple of block height and data root. It is analogous
+to a Celestia block header. `DataRootTuple`s are relayed in batches, committed
+to as a `DataRootTuple`s root (i.e. a Merkle root of `DataRootTuple`s).
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:72]:
```suggestion
to as a `DataRootTupleRoot` (i.e. a Merkle root of `DataRootTuple`s).
```

File: docs/developers/blobstream-contracts.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+The `BinaryMerkleProof` is an [RFC-6962](https://www.rfc-editor.org/rfc/rfc6962.html)-compliant
+Merkle proof. Since `DataRootTuple`s are Merkleized in a binary Merkle tree,
+verifying the inclusion of a `DataRootTuple` against a `DataRootTuple`s root
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:76]: I guess it's better to call it by its name: 

```suggestion
verifying the inclusion of a `DataRootTuple` against a `DataRootTupleRoot` 
```

File: docs/developers/blobstream-contracts.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+allows L2 contracts on Ethereum to query the Blobstream contract for relayed
+`DataRootTuple`s. The single interface method `verifyAttestation` verifies a
+Merkle inclusion proof that a `DataRootTuple` is included under a specific
+batch (indexed by batch nonce). In other words, analogously it verifies that a
+specific block header is included in the canonical Celestia chain.
+
+## Verifying Data Inclusion for Fraud Proofs
+
+A high-level overview of how a fraud-proof based L2 would interact with
+Blobstream can be found [here](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md).
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:91]:
```suggestion
Blobstream can be found in the [inclusion proofs documentation](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md).
```

File: docs/developers/blobstream-contracts.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+The `DAVerifier` library is available at `blobstream-contracts/lib/verifier/DAVerifier.sol`,
+and provides functions to verify the inclusion of individual (or multiple)
+shares against a `DataRootTuple`. The library is stateless, and assumes the
+`DataRootTuple` was previously verified as included through the Blobstream
+contract's `verifyAttestation` method.
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:97]:
```suggestion
shares against a `DataRootTupleRoot`. The library is stateless, and allows to pass an
`IDAOracle` interface as a parameter to verify inclusion against it.
```

File: docs/developers/blobstream-contracts.md (hunk 1 above)
Comment (adlerjohn) [reply docs/developers/blobstream-contracts.md:64]: No, it's just a simple ZK rollup.

File: docs/developers/blobstream-contracts.md (hunk 2 above)
Comment (adlerjohn) [reply docs/developers/blobstream-contracts.md:72]: I considered that originally, but `DataRootTupleRoot` isn't a type, it's a variable name!

File: docs/developers/blobstream.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+Blobstream is the first data availability solution for Ethereum that securely
+scales with the number of users. Formerly known as the [Quantum Gravity Bridge (QGB)](https://blog.celestia.org/celestiums/),
+Blobstream relays commitments to Celestia's data root to an onchain light client
+on Ethereum, for integration by developers into L2 contracts. This enables Ethereum
+developers to build high-throughput L2s using Celestia's optimised DA layer,
+the first with Data Availability Sampling (DAS).
+
+The following docs go over how developers can integrate Blobstream. There are also docs on
+how to run a [Blobstream orchestrator](../nodes/blobstream-intro.md) as a Celestia validator
```
Comment (jcstein) [inline docs/developers/blobstream.md:18]:
```suggestion
how to run a [Blobstream orchestrator-relayer](../nodes/blobstream-intro.md) as a Celestia validator
```

or "orchestrator and relayer" -- as there are links on the `blobstream-intro.md` page for both orchestrator and relayer

File: docs/developers/blobstream-contracts.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+library is available at `blobstream-contracts/lib/verifier/DAVerifier.sol`, and
+provides functions to verify the inclusion of individual (or multiple) shares
+against a `DataRootTuple`. The library is stateless, and assumes the
+`DataRootTuple` was previously verified as included through the Blobstream
+contract's `verifyAttestation` method.
+
+In the `DAVerifier` library, we find functions that help
+with data inclusion verification and calculating the square size of a
+Celestia block. These functions work with the Blobstream smart contract,
+using different proofs to check and confirm the data's validity. Let's
```
Comment (SweeXordious) [inline docs/developers/blobstream-contracts.md:117]:
```suggestion
using different proofs to check and confirm the data's validity. Let's
```
*validity* or *availability* ? we only prove that the data was posted, parsing the data and making sure it's valid is rollup specific, no?

**Review by SweeXordious — COMMENTED** [review]
:fire:

File: docs/developers/blobstream-contracts.md (hunk 7 above)
Comment (jcstein) [reply docs/developers/blobstream-contracts.md]: yes! updated this in 5f23d0011c798e3ac37f3b1e6deb3f3732ebef67

File: docs/developers/blobstream-contracts.md (hunk 5 above)
Comment (jcstein) [reply docs/developers/blobstream-contracts.md]: 93bfcedd2ecb0feab55b6b2aa637aa7ef03e231a

**Review by jcstein — APPROVED** [review]

File: docs/developers/blobstream.md (hunk 6 above)
Comment (adlerjohn) [reply docs/developers/blobstream.md:18]: IMO just orchestrator, since validators only run the orchestrator. Relayer is a third-party service.

**Comment by musalbas** [comment]
Could we add some pointers of how you generate the proofs under "Verifying data inclusion for fraud proofs" with ie links to the RPC endpoints in https://node-rpc-docs.celestia.org/?

**Comment by SweeXordious** [comment]
https://github.com/celestiaorg/docs/pull/1152#issuecomment-1769527854 done here https://github.com/celestiaorg/docs/pull/1158

**Pull request merged**